code/modules/vehicles/mecha/mech_bay.dm

```diff
@@ -128,23 +196,37 @@
 		return data
 
 	data["recharge_port"] = list("mech" = null)
-	var/obj/vehicle/sealed/mecha/recharging_mech = recharge_port.recharging_mech_ref?.resolve()
 
-	if(!recharging_mech)
+	var/obj/recharging_machine = recharge_port?.recharging_machine_ref?.resolve()
+	if(!recharging_machine)
```

Comment:
Does the UI have a dimmer or something that reports it has no linked recharger port? If not it'd fit well